src/contracts/libraries/BeaconChainProofs.sol

```diff
@@ -281,7 +281,14 @@ library BeaconChainProofs {
         require(proofs.timestampProof.length == 32 * (EXECUTION_PAYLOAD_HEADER_FIELD_TREE_HEIGHT),
             "BeaconChainProofs.verifyWithdrawalProofs: timestampProof has incorrect length");
 
-        {
+        if(proofs.proveHistoricalRoot){
```

Comment:
my opinion is that we only allow proofs against hisorical summaries. not worth maintaining 2 things.

Reply:
I disagree - ok so now they have to wait up to 27hours until they can withdraw? Its not like we're "maintaining" two things.  We just generate an additional proof when you're proving an older root.

Reply:
It's 2 different proof generation codes, no?

Reply:
whats a proof generation code?

Reply:
agreed offline to keep 1 proof